src/App.tsx

```diff
@@ -1,87 +1,96 @@
+  readonly tasks = observable<Task>([]);
+  hideCompleted = false;
+  constructor() {
+    makeAutoObservable(this);
+  }
+  replaceTasks(t: Task[]) {
+    this.tasks.replace(t.map(t => makeAutoObservable({ ...t })));
+  }
+  async loadTasks() {
+    this.replaceTasks(await
+      this.taskRepo.find({
+        limit: 20,
+        orderBy: { completed: "asc" },
+        where: { completed: this.hideCompleted ? false : undefined }
+      }));
+  }
+  addTask() {
+    this.tasks.push(makeAutoObservable(new Task()))
+  }
+  async saveTask(task: Task) {
+    try {
+      const savedTask = await this.taskRepo.save(task);
+      runInAction(() => store.tasks[store.tasks.indexOf(task)] = savedTask);
+    } catch (error: any) {
+      alert(error.message);
+    }
+  }
+  async deleteTask(task: Task) {
+    await this.taskRepo.delete(task);
+    runInAction(() => store.tasks.remove(task));
+  }
+  async setAll(completed: boolean) {
+    await TasksController.setAll(completed);
+    this.loadTasks();
+  }
+}
+const store = new Store();
 
 function App() {
-  const [tasks, setTasks] = useState<Task[]>([]);
-  const [hideCompleted, setHideCompleted] = useState(false);
+  return <Todo store={store} />
 
+}
+const Todo: React.FC<{ store: Store }> = observer(({ store }) => {
   useEffect(() => {
-    fetchTasks(hideCompleted).then(setTasks);
-  }, [hideCompleted]);
-
-  const addTask = () => {
-    setTasks([...tasks, new Task()])
-  };
-  const setAll = async (completed: boolean) => {
-    await TasksController.setAll(completed);
-    setTasks(await fetchTasks(hideCompleted));
-  }
-
+    store.loadTasks()
+  }, [store.hideCompleted]);
   return (
     <div>
       <input
         type="checkbox"
-        checked={hideCompleted}
-        onChange={e => setHideCompleted(e.target.checked)} /> Hide Completed
+        checked={store.hideCompleted}
+        onChange={action(e => store.hideCompleted = e.target.checked)} /> Hide Completed
       <main>
-        {tasks.map(task => {
-          const handleChange = (values: Partial<Task>) => {
-            setTasks(tasks.map(t => t === task ? { ...task, ...values } : t));
-          };
-
-          const saveTask = async () => {
-            try {
-              const savedTask = await taskRepo.save(task);
-              setTasks(tasks.map(t => t === task ? savedTask : t));
-            } catch (error: any) {
-              alert(error.message);
-            }
-          }
-
-
-          const deleteTask = async () => {
-            await taskRepo.delete(task);
-            setTasks(tasks.filter(t => t !== task));
-          };
-
+        {store.tasks.map(task => {
           return (
             <div key={task.id}>
               <input type="checkbox"
                 checked={task.completed}
-                onChange={e => handleChange({ completed: e.target.checked })} />
+                onChange={action(e => {
+                  const prev = task.completed;
+                  task.completed = e.target.checked;
```

Comment:
Same comment about passing actions as props into UI components, making them completely decoupled from the store/mobx/whatever implementation details.